fe/fe-core/src/main/java/org/apache/doris/external/iceberg/IcebergCatalogMgr.java

```diff
@@ -50,83 +49,62 @@ public class IcebergCatalogMgr {
 
     // hive metastore uri -> iceberg catalog
     // used to cache iceberg catalogs
-    private static final ConcurrentHashMap<String, IcebergCatalog> metastoreUriToCatalog = new ConcurrentHashMap();
+    private static final ConcurrentHashMap<IcebergProperty, IcebergCatalog> propertiesToCatalog = new ConcurrentHashMap();
```

Comment:
It's better to use `IcebergProperty.extraProperties`, such as `uri`/`warehouse`, as the cache key. 
`IcebergProperty` is bound to the database and table, not a generic cache.

Reply:
Updated